focus/internals/src/lib/model/selection/project.rs

```diff
@@ -129,10 +136,34 @@ impl TryFrom<&Project> for TargetSet {
 /// ProjectSet is a file-level container for projects.
 #[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
 struct ProjectSet {
-    pub projects: HashSet<Project>,
+    pub projects: Vec<Project>,
+}
+impl ProjectSet {
+    #[allow(dead_code)]
+    pub(crate) fn remove_project(&mut self, project_name: String) -> Result<()> {
+        self.projects = self
+            .projects
+            .clone()
+            .into_iter()
+            .filter(|p| p.name != project_name)
+            .collect();
+        Ok(())
+    }
+    pub(crate) fn set_project(&mut self, project_name: String, project: Project) -> Result<()> {
```

Comment:
This function updates a project if the passed in name matches a project and appends to the projects if there is no match?

Reply:
exactly, i'll add that as a comment.